test/integration/particle_emitter.cc

```diff
@@ -0,0 +1,170 @@
+
+                EXPECT_EQ("smoke_emitter", _name->Data());
+                EXPECT_EQ(_name->Data(), _emitter->Data().name());
+                EXPECT_EQ(msgs::ParticleEmitter_EmitterType_BOX,
+                    _emitter->Data().type());
+                EXPECT_EQ(math::Pose3d(0, 1, 0, 0, 0, 0), _pose->Data());
+                EXPECT_EQ(_pose->Data(),
+                    msgs::Convert(_emitter->Data().pose()));
+                EXPECT_EQ(math::Vector3d(2, 2, 2),
+                    msgs::Convert(_emitter->Data().size()));
+                EXPECT_DOUBLE_EQ(5.0, _emitter->Data().rate());
+                EXPECT_DOUBLE_EQ(1.0, _emitter->Data().duration());
+                EXPECT_TRUE(_emitter->Data().emitting());
+                EXPECT_EQ(math::Vector3d(3, 3, 3),
+                    msgs::Convert(_emitter->Data().particle_size()));
+                EXPECT_DOUBLE_EQ(2.0, _emitter->Data().lifetime());
+                // TODO(anyone) add material check here
+                EXPECT_DOUBLE_EQ(10.0, _emitter->Data().min_velocity());
+                EXPECT_DOUBLE_EQ(20.0, _emitter->Data().max_velocity());
+                EXPECT_EQ(math::Color::Blue,
+                    msgs::Convert(_emitter->Data().color_start()));
+                EXPECT_EQ(math::Color::Green,
+                    msgs::Convert(_emitter->Data().color_end()));
+                EXPECT_DOUBLE_EQ(10.0, _emitter->Data().scale_rate());
+
+                // color range image is empty because the emitter system
+                // will not be able to find a file that does not exist
+                // TODO(anyone) this should return  "/path/to/dummy_image.png"
+                // and let rendering do the findFile instead
+                EXPECT_EQ(std::string(),
+                    _emitter->Data().color_range_image());
+              }
+              else
+              {
+                updateDefaultChecked = true;
+
+                EXPECT_TRUE(_name->Data().find(std::to_string(_entity))
+                    != std::string::npos);
+                EXPECT_EQ(_name->Data(), _emitter->Data().name());
+                EXPECT_EQ(msgs::ParticleEmitter_EmitterType_POINT,
+                    _emitter->Data().type());
+                EXPECT_EQ(math::Pose3d(0, 0, 0, 0, 0, 0), _pose->Data());
+                EXPECT_EQ(_pose->Data(),
+                    msgs::Convert(_emitter->Data().pose()));
+                EXPECT_EQ(math::Vector3d(1, 1, 1),
+                    msgs::Convert(_emitter->Data().size()));
+                EXPECT_DOUBLE_EQ(10.0, _emitter->Data().rate());
+                EXPECT_DOUBLE_EQ(0.0, _emitter->Data().duration());
+                EXPECT_FALSE(_emitter->Data().emitting());
+                EXPECT_EQ(math::Vector3d(1, 1, 1),
+                    msgs::Convert(_emitter->Data().particle_size()));
+                EXPECT_DOUBLE_EQ(5.0, _emitter->Data().lifetime());
+                // TODO(anyone) add material check here
```

Comment:
I wasn't sure how to load/test for materials, so I left this check out. If anyone wants to add this check in before merging, that'd be great, but if not, we can do this later.

Reply:
Leaving it for later.